### ISSUE TITLE
hack/e2e.go event stream busted after pflag merge

### ISSUE THREAD

**Issue by zmerlynn** [GoogleCloudPlatform/kubernetes#5902]
This is the script that runs to capture event output during e2e runs: https://github.com/GoogleCloudPlatform/kubernetes/blob/master/hack/e2e-internal/e2e-watch-events.sh

It's failing after the recent pflag merge:
```
watchEvents output: |
  Project: kubernetes-jenkins
  Zone: us-central1-f
  current-context: "kubernetes-jenkins_e2e-test-jenkins"
  Running: /jenkins-master-data/jobs/kubernetes-e2e-gce/workspace/kubernetes/cluster/../cluster/gce/../../cluster/../platforms/linux/amd64/kubectl --match-server-version --watch-only get events
  unknown flag: --watch-only
  
  Usage: 
    kubectl [flags]
    kubectl [command]
  
  Available Commands: 
    version       Print the client and server version information.
    apiversions   Print available API versions.
    clusterinfo   Display cluster info
    proxy         Run a proxy to the Kubernetes API server
    get           Display one or many resources
    describe      Show details of a specific resource
    create        Create a resource by filename or stdin
    update        Update a resource by filename or stdin.
    delete        Delete a resource by filename, stdin, or resource and ID.
    config        config modifies .kubeconfig files
    namespace     SUPERCEDED: Set and view the current Kubernetes namespace
    log           Print the logs for a container in a pod.
    rollingupdate Perform a rolling update of the given ReplicationController.
    resize        Set a new size for a Replication Controller.
    exec          Execute a command in a container.
    port-forward  Forward 1 or more local ports to a pod.
    run-container Run a particular image on the cluster.
    stop          Gracefully shut down a resource by id or filename.
    expose        Take a replicated application and expose it as Kubernetes Service
    label         Update the labels on a resource
    help          Help about any command
  
  Flags:
        --alsologtostderr=false: log to standard error as well as files
        --api-version="": The API version to use when talking to the server
    -a, --auth-path="": Path to the auth info file. If missing, prompt the user. Only used if using https.
        --certificate-authority="": Path to a cert. file for the certificate authority.
        --client-certificate="": Path to a client key file for TLS.
        --client-key="": Path to a client key file for TLS.
        --cluster="": The name of the kubeconfig cluster to use
        --context="": The name of the kubeconfig context to use
    -h, --help=false: help for kubectl
        --insecure-skip-tls-verify=false: If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure.
        --kubeconfig="": Path to the kubeconfig file to use for CLI requests.
        --log_backtrace_at=:0: when logging hits line file:N, emit a stack trace
        --log_dir=: If non-empty, write log files in this directory
        --log_flush_frequency=5s: Maximum number of seconds between log flushes
        --logtostderr=true: log to standard error instead of files
        --match-server-version=false: Require server version to match client version
        --namespace="": If present, the namespace scope for this CLI request.
        --password="": Password for basic authentication to the API server.
    -s, --server="": The address and port of the Kubernetes API server
        --stderrthreshold=2: logs at or above this threshold go to stderr
        --token="": Bearer token for authentication to the API server.
        --user="": The name of the kubeconfig user to use
        --username="": Username for basic authentication to the API server.
        --v=0: log level for V logs
        --validate=false: If true, use a schema to validate the input before sending it
        --vmodule=: comma-separated list of pattern=N settings for file-filtered logging
  
  
  Use "kubectl help [command]" for more information about a command.
  
```

**Comment by zmerlynn** [comment]
cc @jlowdermilk @brendanburns

**Comment by zmerlynn** [comment]
Or something. I might be completely off. But something broke it recently enough, and I know @brendanburns like his event log.

**Comment by jlowdermilk** [comment]
That command is actually invalid. `--watch-only` is a flag on the `get` command, not the root. Fix in #5925
